crates/kitsune_p2p/mdns/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+      let mut substrs = Vec::new();
+      while b64.len() > MAX_TXT_SIZE {
+         let start: String = b64.drain(..MAX_TXT_SIZE).collect();
+         substrs.push(start);
+      };
+      substrs.push(b64);
+      let txts: Vec<_> = substrs.iter().map(AsRef::as_ref).collect();
+      println!("Entering mdns broadcasting thread...");
+      // Create mdns responder
+      let responder = libmdns::Responder::new().unwrap();
+      let _svc = responder.register(
+         HC_SERVICE_TYPE.to_owned(),
+         service_name,
+         0,
+         &txts,
+      );
+      // Loop forever unless termination command received
+      loop {
+         tokio::time::delay_for(::std::time::Duration::from_secs(BROADCAST_INTERVAL_SEC)).await;
+         if !can_run_clone.load(Ordering::Relaxed) {
+            println!("Terminating.");
+            break;
+         }
+      }
+   });
+   // Done
+   can_run
+}
+
+///
+#[derive(Debug, Clone)]
+pub struct MdnsResponse {
+   /// Service name used
+   pub service_name: String,
+   /// IP address that responded to the mdns query
+   pub addr: std::net::IpAddr,
+   /// Data contained in the TXT record
+   pub buffer: Vec<u8>,
+}
+
+/// Queries the network for the holochain service
+/// Returns an iterator over all responses received
+pub fn mdns_listen() -> impl Stream<Item = Result<MdnsResponse, MdnsError>> {
+   let service_name = format!("{}.local", HC_SERVICE_TYPE);
+   let query = mdns::discover::all(service_name, Duration::from_secs(QUERY_INTERVAL_SEC))
+   .expect("mdns Discover failed");
+   // Get Mdns Response stream
+   let response_stream = query.listen();
+   // Change it into a MdnsResponse stream
+   let mdns_stream = response_stream
+      // Filtering out Empty responses
+      .filter(move |res| {
+         futures_util::future::ready(match res {
+            Ok(response) => !response.is_empty() && !response.ip_addr().is_none(),
+            Err(_) => true, // Keep errors
+         })
+      })
+      .map(|maybe_response| {
+         if let Err(e) = maybe_response {
+            return Err(MdnsError::Mdns(e));
+         }
+         let response = maybe_response.unwrap();
+         // NOTE: if response.ip_addr() is not te right address,
+         // we should give all A/AAA records found in the answers instead
+         let addr = response.ip_addr().unwrap(); // should have already been filtered out
+         let mut buffer = Vec::new();
+         let mut service_name = String::new();
+         println!("Response Answer count = {}", response.answers.len());
+         //println!("Response Answers:  {:?}", response.answers);
+         for answer in response.answers {
+            match  answer.kind {
+               RecordKind::TXT(txts) => {
+                  println!("TXT count = {}", txts.len());
+                  let mut b64 = String::new();
+                  for txt in txts {
+                     //println!("Response TXT = {:?}", txt);
+                     b64.push_str(&txt);
```

Comment:
:+1: - Hopefully the txts always come back in the same order : )

Reply:
From my testing it does. If it doesnt on some setups, I could easily prepend an index.